frontend/src/lib/components/QuestionOptions.svelte

```diff
@@ -8,37 +11,105 @@
     // Internal State
     let selected_answer: number[] = [];
     let show_feedback: boolean = false;
+    let correct: number[] = [];
+    let incorrect: number[] = [];
+
+    // this function gets the selected items from the store
+    selected_store.subscribe((_store) => {
+        selected_answer = _store;
+    });
 
+    // this function handles a submit
     function check_answer() {
         console.log(selected_answer);
         return _.isEmpty(_.xor(question.correctAnswer, selected_answer))
-            ? console.log('right')
-            : (show_feedback = true);
+            ? (correct = selected_answer)
+            : ((show_feedback = true), (incorrect = selected_answer));
     }
 </script>
 
 <!-- We can add different types of question options 
     (like matching questions or fill in the blank) here in the future -->
+
 <form on:submit={check_answer}>
     {#each question.options as option, index}
-        {#if question.correctAnswer.length > 1}
-            <input
-                type="checkbox"
-                id={index.toString()}
-                bind:group={selected_answer}
-                value={index}
-            />
+        <!-- this div is displayed if the form has been submitted and the answer is correct -->
+        {#if correct.includes(index)}
+            <div class="m-3 w-1/2 rounded-lg bg-green-200 p-3">
+                <DisplayOption {option} {index} correct={question.correctAnswer} checked={false} />
+            </div>
+
+            <!-- this div is displayed if the form has been submitted and the answer is incorrect -->
+        {:else if incorrect.includes(index)}
+            <div class="m-3 w-1/2 rounded-lg bg-red-200 p-3">
+                <DisplayOption {option} {index} correct={question.correctAnswer} checked={false} />
+            </div>
+
+            <!-- this div is displayed if the form has not been submitted and the option is selected -->
+        {:else if selected_answer.includes(index)}
+            <div class="m-3 w-1/2 rounded-lg bg-slate-300 p-3">
+                <DisplayOption {option} {index} correct={question.correctAnswer} checked={true} />
+            </div>
+
+            <!-- this div is displayed if the form has not been submitted and the option is not selected yet -->
```

Comment:
Really solid code! The comments and different use cases make sense to me at first glance.